cwms_radar_api/src/main/java/cwms/radar/api/RatingSpecController.java

```diff
@@ -0,0 +1,184 @@
+package cwms.radar.api;
+
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletResponse;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import cwms.radar.api.errors.RadarError;
+import cwms.radar.data.dao.JooqDao;
+import cwms.radar.data.dao.RatingSpecDao;
+import cwms.radar.data.dto.rating.RatingSpec;
+import cwms.radar.data.dto.rating.RatingSpecs;
+import cwms.radar.formatters.ContentType;
+import cwms.radar.formatters.Formats;
+import io.javalin.apibuilder.CrudHandler;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+
+public class RatingSpecController implements CrudHandler {
+    private static final Logger logger = Logger.getLogger(RatingSpecController.class.getName());
+    private final MetricRegistry metrics;
+
+    private static final int DEFAULT_PAGE_SIZE = 100;
+
+    private final Histogram requestResultSize;
+
+    public RatingSpecController(MetricRegistry metrics){
+        this.metrics=metrics;
+        String className = this.getClass().getName();
+        requestResultSize = this.metrics.histogram((name(className,"results","size")));
+    }
+
+    protected DSLContext getDslContext(Context ctx)
+    {
+        return JooqDao.getDslContext(ctx);
+    }
+
+    private Timer.Context markAndTime(String subject)
+    {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+    
+    
+    @OpenApi(
+        queryParams = {
+            @OpenApiParam(name="office", description="Specifies the owning office of the Rating Specs whose data is to be included in the response. If this field is not specified, matching rating information from all offices shall be returned."),
+            @OpenApiParam(name="rating-id-mask", description="RegExp that specifies the rating IDs to be included in the response. If this field is not specified, all Rating Specs shall be returned."),
+                @OpenApiParam(name="page",
+                        description = "This end point can return a lot of data, this identifies where in the request you are. This is an opaque value, and can be obtained from the 'next-page' value in the response."
+                ),
+                @OpenApiParam(name="page-size", type=Integer.class,
+                        description = "How many entries per page returned. Default " + DEFAULT_PAGE_SIZE + "."
+                ),
+        },
+            responses = {
```

Comment:
Indentation level doesn't match the surrounding items.
[Line 116](https://github.com/USACE/cwms-radar-api/pull/164/files#diff-b486a96bd450422a047bbf500f6060f007d6052277965934308356e397d951c2R116) is a good reference.

Nitpick, feel free to ignore. :-)